doc/SONiC_static_route_hdl.md

```diff
@@ -0,0 +1,422 @@
+# Static IP route Configuration
+
+Implement Static IP route Configuration via  CLI/REST/gNMI  in SONiC management framework for non-management routes.
+
+# High Level Design Document
+#### Rev 0.1
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author                | Change Description                |
+|:---:|:-----------:|:---------------------------:|-----------------------------------|
+| 0.1 | 03/18/2020  |  Sucheta Mahara             | initial draft                     |
+|  0.2| 03/20/20202 |  Venkatesan Mahalinga       | draft                             |
+| 0.2 | 03/23/2020  |  Zhenhong Zhao              | FRR Config Support             |
+
+# About this Manual
+This document provides general information about configuring static routes via Management CLI/REST/gNMI in Sonic.
+# Scope
+Covers general design for supporting static routes in SONiC Management framework.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| IP                     |  Internet Protocol       |
+| VRF                     | Virtual routing and forwarding
+
+
+# 1 Feature Overview
+
+The documents covers the various interfaces for configuring static routes using SONiC management framework.
+
+## 1.1 Requirements
+
+### 1.1.1 Functional Requirements
+
+Provide ability to configure IPv4 and IPv6 static routes using SONiC management framework.
+
+### 1.1.2 Configuration and Management Requirements
+ - Implement Static ip route CLI Commands.
+ - REST set/get support for Static IPv4 and IPv6 routes.
+ - gNMI set/get support for Static IPv4 and IPv6 routes.
+
+### 1.1.3 Scalability Requirements
+### 1.1.4 Warm Boot Requirements

```

Comment:
May be we should add statement saying.."With static routes configured, system should be able to do warmboot", instead of leaving these sections empty

Reply:
sure